asv_bench/benchmarks/algorithms.py

```diff
@@ -16,56 +16,75 @@
 
-    def setup(self, sort):
+    def setup(self, sort, dtype):
         N = 10**5
-        self.int_idx = pd.Int64Index(np.arange(N).repeat(5))
-        self.float_idx = pd.Float64Index(np.random.randn(N).repeat(5))
-        self.string_idx = tm.makeStringIndex(N)
+        data = {'int': pd.Int64Index(np.arange(N).repeat(5)),
+                'uint': pd.UInt64Index(np.arange(N).repeat(5)),
+                'float': pd.Float64Index(np.random.randn(N).repeat(5)),
+                'string': tm.makeStringIndex(N).repeat(5)}
+        self.idx = data[dtype]
 
-    def time_factorize_int(self, sort):
-        self.int_idx.factorize(sort=sort)
+    def time_factorize(self, sort, dtype):
+        self.idx.factorize(sort=sort)
 
-    def time_factorize_float(self, sort):
-        self.float_idx.factorize(sort=sort)
 
-    def time_factorize_string(self, sort):
-        self.string_idx.factorize(sort=sort)
+class FactorizeUnique(object):
 
+    params = [[True, False], ['int', 'uint', 'float', 'string']]
+    param_names = ['sort', 'dtype']
 
-class Duplicated(object):
+    def setup(self, sort, dtype):
+        N = 10**5
+        data = {'int': pd.Int64Index(np.arange(N)),
+                'uint': pd.UInt64Index(np.arange(N)),
+                'float': pd.Float64Index(np.arange(N)),
+                'string': tm.makeStringIndex(N)}
+        self.idx = data[dtype]
+        assert self.idx.is_unique
 
-    params = ['first', 'last', False]
-    param_names = ['keep']
+    def time_factorize(self, sort, dtype):
+        self.idx.factorize(sort=sort)
 
-    def setup(self, keep):
-        N = 10**5
-        self.int_idx = pd.Int64Index(np.arange(N).repeat(5))
-        self.float_idx = pd.Float64Index(np.random.randn(N).repeat(5))
-        self.string_idx = tm.makeStringIndex(N)
 
-    def time_duplicated_int(self, keep):
-        self.int_idx.duplicated(keep=keep)
+class Duplicated(object):
+
+    params = [['first', 'last', False], ['int', 'uint', 'float', 'string']]
+    param_names = ['keep', 'dtype']
 
-    def time_duplicated_float(self, keep):
-        self.float_idx.duplicated(keep=keep)
+    def setup(self, keep, dtype):
+        N = 10**5
+        data = {'int': pd.Int64Index(np.arange(N).repeat(5)),
+                'uint': pd.UInt64Index(np.arange(N).repeat(5)),
+                'float': pd.Float64Index(np.random.randn(N).repeat(5)),
+                'string': tm.makeStringIndex(N).repeat(5)}
+        self.idx = data[dtype]
+        # cache is_unique
+        self.idx.is_unique
 
-    def time_duplicated_string(self, keep):
-        self.string_idx.duplicated(keep=keep)
+    def time_duplicated(self, keep, dtype):
+        self.idx.duplicated(keep=keep)
 
 
 class DuplicatedUniqueIndex(object):
 
-    def setup(self):
-        N = 10**5
-        self.idx_int_dup = pd.Int64Index(np.arange(N * 5))
+    params = ['int', 'uint', 'float', 'string']
+    param_names = ['dtype']
+
+    def setup(self, dtype):
+        N = 10**5 * 5
```

Comment:
What was the reason for changing this line?

Reply:
Intent was to make `Duplicated` and `DuplicatedUniqueIndex` comparable in terms of `N` (the former is repeated 5x). I'll remove it as it's confusing.

Reply:
Done